src/Parsing/ParserState.php

```diff
@@ -375,27 +375,26 @@ public function consumeExpression($mExpression, $iMaxLength = null): string
      */
     public function consumeComment()
     {
-        if ($this->peek() != '/' || $this->peek(1, 1) != '*') {
-            return false;
-        }
-
-        $sComment = '';
-        $iLineNo = $this->iLineNo;
-        $this->consume(1);
-        while (($char = $this->consume(1)) !== '') {
-            if ($char == '*' && $this->peek() == '/') {
-                $this->consume(1);
-                break;
+        $mComment = false;
+        if ($this->comes('/*')) {
+            $iLineNo = $this->iLineNo;
+            $this->consume(1);
+            $mComment = '';
+            while (($char = $this->consume(1)) !== '') {
+                $mComment .= $char;
+                if ($this->comes('*/')) {
+                    $this->consume(2);
+                    break;
+                }
             }
-            $sComment .= $char;
         }
 
-        if ($sComment !== '') {
+        if ($mComment !== false) {
```

Comment:
Maybe we should also take into account an empty string as a possible value?

If we're going with the early return approach, we could just check for an empty string here.

```suggestion
        if ($mComment !== '') {
```